share/eds/store.go

```diff
@@ -71,23 +81,48 @@ func NewStore(basepath string, ds datastore.Batching) (*Store, error) {
 
 	return &Store{
 		basepath: basepath,
-		dgstr:    dagStore,
-		topIdx:   invertedRepo,
-		carIdx:   fsRepo,
-		mounts:   r,
+		// initialize empty gc result to avoid panic on access
+		lastGCResult: &dagstore.GCResult{
+			Shards: make(map[shard.Key]error),
+		},
+		dgstr:  dagStore,
+		topIdx: invertedRepo,
+		carIdx: fsRepo,
+		mounts: r,
 	}, nil
 }
 
-// Start starts the underlying DAGStore.
 func (s *Store) Start(ctx context.Context) error {
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+	go s.gc(s.ctx)
+
 	return s.dgstr.Start(ctx)
 }
 
-// Stop stops the underlying DAGStore.
 func (s *Store) Stop() error {
```

Comment:
We usually pass context to  `Stop()`, let's do it here too.

```suggestion
func (s *Store) Stop(context.Context) error {
```

Reply:
We usually do it because FX requires us, but with its recent features, it no more does this, so we can stop passing context where  unnecessary